.github/workflows/nim-matrix.yml

```diff
@@ -0,0 +1,30 @@
+name: Nim matrix
+
+on:
+  merge_group:
+  workflow_dispatch:
+
+env:
+  cache_nonce: 0 # Allows for easily busting actions/cache caches
+  nim_version: v1.6.14, v1.6.16, v1.6.18, v2.0.0, v2.0.2
```

Comment:
Guys, why do we have 2.x in the compiler matrix? Is Codex supposed to work with 2.x?

Reply:
yes, we want to start moving to 2.x

Reply:
oh... but this will fail for now, since we haven't completed the move @veaceslavdoina

Reply:
That was [the idea](#issuecomment-1928398091)?
> We probably should enable merge queue only after confirmation that builds with Nim v2 passes. We can trigger Nim-matrix workflow manually until that.

Merge queue is not enabled and `Nim-matrix` workflow is not used yet, may be just to test how it works. If we want to use it right now, we then should remove v2 from the list.

Should we?

Reply:
Ah I see, well, if it isn't being used for now, then no harm in keeping it, but as soon as we do, it will break with 2.x and so it would need to be disabled until we port Codex over.